application/controllers/Docs.php

```diff
@@ -30,67 +30,35 @@ function __construct() {
 	public function index($page = 'main')
 	{
 
-		$data = array();
-
-		$docs_path = ($this->config->item('docs_path')) ? $this->config->item('docs_path') : 'https://raw.githubusercontent.com/GSA/project-open-data-dashboard/master/documentation/';
-		$docs_path = $docs_path	. $page . '.md';
-		$docs = @file_get_contents($docs_path);
-
-		if($docs) {
-
-			$markdown_extra = new Michelf\MarkdownExtra();
-
-			$markdown_text = $docs;
-
-			$markdown_text = $markdown_extra->transform($markdown_text);
-			$markdown_text = linkToAnchor($markdown_text);
-
-			$data['docs_html'] = $markdown_text;
-
-		} else {
-			$data['docs_html'] = "The documentation file is unavailable";
-		}
-
+        // Docs live parallel to the app itself
+        $docsPath = realpath(APPPATH . '..').'/documentation/';
+
+        // Check that the page requested is one we actually have
+        $docFiles = glob($docsPath.'*.md');
+        foreach ($docFiles as $filename) {
+            $availableDocs[] = basename($filename, '.md');
+        }
+        if (!in_array($page, $availableDocs)) {
+            show_404($page . ' documentation file is unavailable');
```

Comment:
Since `$page` is untrusted content, we should be careful it's being escaped properly when displayed. On first glance, this _looks_ like it could be used for XSS but it's not actually rendered in the 404 page. In this case, we're not using a custom 404 template, so it's not rendered at all.

Also, looking at the [docs](https://codeigniter.com/userguide3/general/errors.html?highlight=show_404#show_404), this parameter should be a URL, not a message.

I don't know if there's any action to take at this point, but it made me look twice when I read it.

Maybe we should open an issue about XSS in templates and document this case :shrug: